src/core/elf.h

```diff
@@ -3,43 +3,250 @@
+#define ELF_DYNAMIC_TABLE_TYPE_STRSZ    10 //Size, in bytes, of the string table (val)
+#define ELF_DYNAMIC_TABLE_TYPE_SYMENT   11 //Size, in bytes, of each symbol entry (val)
+#define ELF_DYNAMIC_TABLE_TYPE_INIT     12 //Address of the initialization function (ptr)
+#define ELF_DYNAMIC_TABLE_TYPE_FINI     13 //Address of the termination function (ptr)
+#define ELF_DYNAMIC_TABLE_TYPE_SONAME   14 //String table offset to the name of this shared object (val)
+#define ELF_DYNAMIC_TABLE_TYPE_RPATH    15 //String table offset to the shared library search path string (val)
+
+//The presence of this dynamic table entry modifies the
+//symbol resolution algorithm for references within the
+//library. Symbols defined within the library are used to
+//resolve references before the dynamic linker searches the
+//usual search path.
+#define ELF_DYNAMIC_TABLE_TYPE_SYMBOLIC 16
+
+#define ELF_DYNAMIC_TABLE_TYPE_REL      17 //Address of a relocation table with Rel entries (ptr)
+#define ELF_DYNAMIC_TABLE_TYPE_RELSZ    18 //Size, in bytes, of the Rel relocation table (val)
+#define ELF_DYNAMIC_TABLE_TYPE_RELENT   19 //Size, in bytes, of each Rel relocation entry (val)
+#define ELF_DYNAMIC_TABLE_TYPE_PLTREL   20 //Type of relocation entry used for the procedure linkage table. The val member contains either TABLE_TYPE_REL or TABLE_TYPE_RELA.
+#define ELF_DYNAMIC_TABLE_TYPE_DEBUG    21 //Reserved for debug usage (ptr)
+
+//The presence of this dynamic table entry signals that the
+//relocation table contains relocations for a non-writable
+//segment.
+#define ELF_DYNAMIC_TABLE_TYPE_TEXTREL  22
+
+#define ELF_DYNAMIC_TABLE_TYPE_JMPREL   23 //Address of the relocations associated with the procedure linkage table (ptr)
+
+//The presence of this dynamic table entry signals that the
+//dynamic loader should process all relocations for this object
+//before transferring control to the program.
+#define ELF_DYNAMIC_TABLE_TYPE_BIND_NOW 24
+
+#define ELF_DYNAMIC_TABLE_TYPE_INIT_ARRAY   25 //Pointer to an array of pointers to initialization functions (ptr)
+#define ELF_DYNAMIC_TABLE_TYPE_FINI_ARRAY   26 //Pointer to an array of pointers to termination functions (ptr)
+#define ELF_DYNAMIC_TABLE_TYPE_INIT_ARRAYSZ 27 //Size, in bytes, of the array of initialization functions (val)
+#define ELF_DYNAMIC_TABLE_TYPE_FINI_ARRAYSZ 28 //Size, in bytes, of the array of termination functions (val)
+#define ELF_DYNAMIC_TABLE_TYPE_LOOS         0x60000000 //Defines a range of dynamic table tags that are reserved for environment-specific use
+#define ELF_DYNAMIC_TABLE_TYPE_HIOS         0x6FFFFFFF //Defines a range of dynamic table tags that are reserved for environment-specific use
+#define ELF_DYNAMIC_TABLE_TYPE_LOPROC       0x70000000 //Defines a range of dynamic table tags that are reserved for processor-specific use.
+#define ELF_DYNAMIC_TABLE_TYPE_HIPROC       0x7FFFFFFF //Defines a range of dynamic table tags that are reserved for processor-specific use.
```

Comment:
Do we need all these constants? If not, I suggest only adding the ones we really need. That's a lot of information and I'd like to maintain a healthy project :p

Reply:
Please see my previous comment: 
```
I thought it'd be a good idea to have all the details about the ELF64 spec. We could move them inside the .c instead, since they're not of much use otherwise.
```